src/include/sof/audio/buffer.h

```diff
@@ -72,6 +76,11 @@ struct comp_buffer {
 	struct list_item source_list;	/* list in comp buffers */
 	struct list_item sink_list;	/* list in comp buffers */
 
+	/* callbacks */
+	void (*cb)(void *data, uint32_t bytes);
```

Comment:
whats the usecase for the cb ? Can you give an example ?

Reply:
Action that we want to take place right after buffer will be read/written. E.g. host can copy new data from Host DMA right after it makes sure, that the previous data has been already consumed. In the same time we don't need to wait for the next copy and data is always "fresh" and ready.

Reply:
So this pushes component level processing functionality into buffers (which should be generic/dumb). This should really be in the component, the above example could be handled in copy() or IRQ handler ?

Reply:
It is generic, because you can do whatever you want in the callbacks. I've just outlined the scenario for what I'm using it. Initially it was done in the copy, but it doesn't scale up towards different scheduling modes. When we are operating on the ping-pong buffer and have no realtime control over DMAs (especially DW-DMA), then it is hard to keep up with data copying and in the same time synchronizing write/read pointers on both sides.